src/jenkins.py

```diff
@@ -144,6 +146,467 @@ def _get_api_credentials(container: ops.Container) -> Credentials:
         raise JenkinsBootstrapError("Admin API credentials not yet setup.") from exc
 
 
+class Jenkins:
+    """Wrapper for Jenkins functionality.
+
+    Attrs:
+        environment: the Jenkins environment configuration.
+        web_url: the Jenkins web URL.
+        login_url: the Jenkins login URL.
+        version: the Jenkins version.
+    """
+
+    environment: Environment
+
+    def __init__(self, environment: Environment):
+        """Construct a Jenkins class.
+
+        Args:
+            environment: the Jenkins environment.
+        """
+        self.environment = environment
+
+    @property
+    def web_url(self) -> str:
+        """Get the Jenkins web URL.
+
+        Returns: the web URL.
+        """
+        env_dict = typing.cast(typing.Dict[str, str], self.environment)
```

Comment:
Isn't this already declared as a dict? What is the reason for needing the cast here please?

Reply:
Added a comment explaining why